### PR TITLE
feat: [FILES-725] handle missing cookies for collaboration links

### PR DESCRIPTION
**Pull request by larcara**
Avoid redirect to login page in case of public link

### CODE REVIEW

**Review by sonersivri — APPROVED** [review]

File: conf/nginx/templates/nginx.conf.web.https.default.template [hunk 1]
```diff
@@ -128,6 +128,18 @@ server
 
     location /services/files/
     {
+        set $public_link 0;
+        set $AUTH_COOKIE_MISSING 0;
+        if ($uri ~* "^/services/files/link/.*$")  {
```
Comment (federicorispo) [inline conf/nginx/templates/nginx.conf.web.https.default.template:133]: With the new public folder access feature we changed the link used to download a file maintaining the backward compatibility with the old link (`/services/files/link`). So you need to accept also the new one: `/services/files/public/link/download/`

File: conf/nginx/templates/nginx.conf.web.https.template [hunk 2]
```diff
@@ -90,6 +90,18 @@ server
 
     location /services/files/
     {
+        set $public_link 0;
+        set $AUTH_COOKIE_MISSING 0;
+        if ($uri ~* "^/services/files/link/.*$")  {
```
Comment (federicorispo) [inline conf/nginx/templates/nginx.conf.web.https.template:95]: See above

**Review by federicorispo — CHANGES_REQUESTED** [review]

**Review by keshavbhatt — CHANGES_REQUESTED** [review]

File: conf/nginx/templates/nginx.conf.web.https.default.template (hunk 1 above)
Comment (larcara) [reply conf/nginx/templates/nginx.conf.web.https.default.template]: now both /service/files/link and /service/files/public are managed as URLs that do not require redirect

File: conf/nginx/templates/nginx.conf.web.https.template (hunk 2 above)
Comment (larcara) [reply conf/nginx/templates/nginx.conf.web.https.template]: now both /service/files/link and /service/files/public are managed as URLs that do not require redirect

File: conf/nginx/templates/nginx.conf.web.https.template [hunk 3]
```diff
@@ -90,6 +90,18 @@ server
...
+        set $public_link 1;
+        }
+        if ($http_cookie !~ "ZM_AUTH_TOKEN=") {
+        set $AUTH_COOKIE_MISSING 1;
+        }
+        set $require_login $public_link:$AUTH_COOKIE_MISSING;
```
[the remark this answers is not in the document]
Comment (larcara) [reply conf/nginx/templates/nginx.conf.web.https.template:101]: @keshavbhatt i tried but it seems is not working. I'll try after the release to clean up the config

**Review by federicorispo — APPROVED** [review]
LGTM :rocket:

**Review by keshavbhatt — APPROVED** [review]

**Pull request merged**